src/connectors/articleService.ts

```diff
@@ -108,33 +109,58 @@ export class ArticleService extends BaseService {
   }
 
   /**
-   * Publish data to IPFS
+   * Publish draft data to IPFS
    */
   publishToIPFS = async ({
     authorId,
     title,
     cover,
-    summary,
     content,
+    circleId,
+    summary,
   }: Record<string, any>) => {
     const userService = new UserService()
     const systemService = new SystemService()
-    const now = new Date()
 
     // prepare metadata
     const author = await userService.dataloader.load(authorId)
-    const { avatar, description, displayName, userName } = author
+    const {
+      avatar,
+      description,
+      displayName,
+      userName,
+      paymentPointer,
+    } = author
     const userImg = avatar && (await systemService.findAssetUrl(avatar))
     const articleImg = cover && (await systemService.findAssetUrl(cover))
 
-    // add content to ipfs
-    const bundle = await makeHtmlBundle({
+    const bundleInfo = {
       title,
       author: { userName, displayName },
       summary,
       content,
       prefix: IPFS_PREFIX,
-    })
+    } as FormatterVars
+
+    // paywall info
+    if (circleId) {
+      const [
+        { name: circleName, displayName: circleDisplayName },
```

Comment:
destructuring will be failed if circleId hasn't query record?

Reply:
Destructuring undefined does cause error.